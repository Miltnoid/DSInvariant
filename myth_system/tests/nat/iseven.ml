type nat =
  | O
  | S of nat

type bool =
  | True
  | False

let div2 : nat -> bool |>
  { 0 => True
  | 1 => False
  | 2 => True
  | 3 => False
  } = ?
