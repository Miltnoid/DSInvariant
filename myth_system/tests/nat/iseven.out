(* Generated Mon Dec 29 00:37:31 2014 *)
let div2 : nat -> bool =
  let rec f1 (n1:nat) : bool =
    match n1 with
      | O -> True
      | S n2 -> (match n2 with
                   | O -> False
                   | S n3 -> f1 n3)
  in
    f1
;;
